dds/DCPS/GuidUtils.h

```diff
@@ -193,17 +193,29 @@ inline void assign(GuidPrefix_t& dest, const GuidPrefix_t& src)
   std::memcpy(&dest[0], &src[0], sizeof(GuidPrefix_t));
 }
 
-inline DCPS::RepoId make_id(const GuidPrefix_t& prefix, const EntityId_t& entity)
+inline void assign(GUID_t& dest, const GUID_t& src)
 {
-  DCPS::RepoId id;
+  std::memcpy(&dest, &src, sizeof(GUID_t));
+}
+
+inline void assign(OctetArray16& dest, const GUID_t& src)
+{
+  std::memcpy(&dest[0], &src.guidPrefix, sizeof(GuidPrefix_t));
```

Comment:
We know that `sizeof(dest)` has to be `sizeof(GUID_t)`.  Change to single memcpy.

Reply:
I know that's how we normally treat it, but `GUID_t` is a struct. Is it guaranteed that it won't have padding inside it?

Reply:
In this case, yes I think so.  Everything in it is struct/array of single bytes.

Reply:
I guess if it's done like in CDR, still seems weird to me to assume how the compiler is going to layout an unpacked struct.

Reply:
I will make the change and make a unit test that asserts the two sizes are the same.